### ISSUE TITLE
Search for members on paginated member page

### ISSUE THREAD

**Issue by aerugo** [Edgeryders-Participio/multi-dreams#369]

**Comment by aerugo** [comment]
related https://github.com/Edgeryders-Participio/multi-dreams/issues/307 /j

**Comment by Powersource** [comment]
fwiw there's some member searching added to the api in https://github.com/cobudget/cobudget/pull/525 that might be reusable/usable for inspiration

**Comment by aerugo** [comment]
Ping @SeunghunSunmoonLee

**Comment by SeunghunSunmoonLee** [comment]
Just followed README and I get error. 
`➜  ui git:(main) ✗ yarn dev
yarn run v1.22.4
$ yarn run generate
$ yarn run generate:prisma
$ prisma generate
Environment variables loaded from .env
Prisma schema loaded from server/prisma/schema.prisma

✔ Generated Prisma Client (3.4.1) to ./node_modules/@prisma/client in 344ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /Users/seunghunlee/Software/cobudget/ui/.env.local
info  - Loaded env from /Users/seunghunlee/Software/cobudget/ui/.env.development
info  - Loaded env from /Users/seunghunlee/Software/cobudget/ui/.env
info  - Disabled SWC as replacement for Babel because of custom Babel configuration ".babelrc" https://nextjs.org/docs/messages/swc-disabled
event - compiled client and server successfully in 2.9s (2113 modules)
wait  - compiling / (client and server)...
info  - Using external babel configuration from /Users/seunghunlee/Software/cobudget/ui/.babelrc
event - compiled client and server successfully in 1483 ms (2116 modules)
wait  - compiling /api...
event - compiled client and server successfully in 431 ms (2160 modules)
wait  - compiling /_error (client and server)...
error - ReferenceError: Cannot access '__WEBPACK_DEFAULT_EXPORT__' before initialization
event - compiled client and server successfully in 292 ms (2161 modules)`

**Comment by Powersource** [comment]
@SeunghunSunmoonLee it's a strange bug since neither Gustav nor I are getting it, but both you and also another person I've heard get it. But it seems to be fixed by https://github.com/vercel/next.js/pull/33831 , however that pr doesn't seem to have ended up in a normal release yet so you might have to update to a canary version of `next`, e.g. `12.0.11-canary.9`

**Comment by aerugo** [comment]
@SeunghunSunmoonLee - to log in with a magic link, you use the link that is printed to terminal when you try to log in with any email.

**Comment by aerugo** [comment]
Did you get it to work @SeunghunSunmoonLee?

**Comment by SeunghunSunmoonLee** [comment]
Yeah. I have WIP solution. Can you let me push? 
```
➜  cobudget git:(369-search-for-members) git push --set-upstream origin 369-search-for-members
remote: Permission to cobudget/cobudget.git denied to SeunghunSunmoonLee.
fatal: unable to access 'https://github.com/cobudget/cobudget.git/': The requested URL returned error: 403
```

**Comment by SeunghunSunmoonLee** [comment]
@Powersource I've used your implementation. I got error with usernameStartsWith. 
```
CombinedError {name: 'CombinedError', message: '[GraphQL] Unknown argument "usernameStartsWith" on field "MembersPage.members".', graphQLErrors: Array(1), networkError: undefined, response: Response}
```
Do you have any clue on this? 

Not sure why MembersPage resolver is reading members from db while members do the same?

**Comment by Powersource** [comment]
@SeunghunSunmoonLee MembersPage is(/was? I think we might have switched to another one in some/all places) our solution for pagination.

For that error, when adding an argument to a graphql resolver you need to add it in a few different places, including in the graphql schema.

**Comment by Powersource** [comment]
i think he's trying to push to a new branch in our repo. that would be fine but he'd need more permissions.

@SeunghunSunmoonLee i think it's simplest if you create a fork on github and push to that one instead

**Comment by SeunghunSunmoonLee** [comment]
Here it is :) https://github.com/cobudget/cobudget/pull/545

**Comment by francesca-sp** [comment]
Following a discussion in today's meeting, I've added this new issue to improve the search options for the round members: https://github.com/cobudget/cobudget/issues/369 ping @SeunghunSunmoonLee

**Comment by Powersource** [comment]
@SeunghunSunmoonLee assigned this to you since iirc you were still gonna work on cobudget, just not very much per month?

**Comment by SeunghunSunmoonLee** [comment]
Hey! Which ticket? #554 or #555?

**Comment by Powersource** [comment]
@SeunghunSunmoonLee yep this issue, i've updated the description a tad to clarify. specifically we want

> Search on Group (aka org) member list page

We currently only have it on the Round (aka collection) page

**Comment by Powersource** [comment]
This issue should be closable now. As mentioned in the PR, new features to test:
* pagination on round members page
* search on group members page
* ability to search on email on the round members list (for admins only)

**Comment by francesca-sp** [comment]
The last 2 bullet points are working, and for pagination, I'm trying to test it - but it's not even kicking in with a list of 27 members. As of how many members is pagination supposed to start? @Powersource

**Comment by gustavlrsn** [comment]
Hm, I only see 3 participants in that list. But unsure what you mean with the load more behavior. Clicking the button should add up to 30 more records in the same list. We're adding automatic "infinite scrolling" in another issue, so that you don't need to press the button manually.

**Comment by francesca-sp** [comment]
that's very odd, there are 31 people in that list. 
![image](https://user-images.githubusercontent.com/3641828/176208113-0fe7e328-a87e-4f8a-9d9e-70f34605e201.png)

My understanding was that this issue is about "pagination", which I understand as pages showing up that you switch between, once you reach above 30. Is that incorrect?

**Comment by gustavlrsn** [comment]
Right, as a non-admin I can only see people that have accepted the invitation. 

Ah, yes, confusing term, it is pagination more in the technical sense of dividing up the result in "pages", to avoid loading everything at once

**Comment by francesca-sp** [comment]
ok, I understand now. So in that case, this seems to be working and can be closed.

**Comment by gustavlrsn** [comment]
Great, closing this
